### PR TITLE
Add AST classes with Bison integration

### LINKED ISSUE #10: Create AST class structure

### PR DESCRIPTION
**Pull request by taylor-santos**
closes #10

### CODE REVIEW

**Comment by taylor-santos** [comment]
@Dampf#2720

**Review by taylor-santos — COMMENTED** [review]
@Crollt

**Review by taylor-santos — COMMENTED** [review]
@Crollt

File: include/ast/declaration.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#ifndef DECLARATION_H
+#define DECLARATION_H
+
+#include "statement.h"
+
+namespace TuataraAST {
+
+class Declaration : public Statement {
+protected:
```
Comment (dampfhub) [inline include/ast/declaration.h:9]: Why protected twice?

**Review by dampfhub — APPROVED** [review]

**Pull request merged**